src/Microsoft.Identity.Client/ApiConfig/AcquireTokenBuilder.cs

```diff
@@ -0,0 +1,136 @@
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.ApiConfig
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public static class AcquireTokenBuilder
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <param name="authorizationCode"></param>
+        /// <returns></returns>
+        public static AcquireTokenByAuthorizationCodeParameterBuilder CreateForAuthorizationCode(
+            IEnumerable<string> scopes,
+            string authorizationCode)
+        {
+            return AcquireTokenByAuthorizationCodeParameterBuilder.Create(scopes, authorizationCode);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <returns></returns>
+        public static AcquireTokenForClientParameterBuilder CreateForClient(IEnumerable<string> scopes)
+        {
+            return AcquireTokenForClientParameterBuilder.Create(scopes);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static AcquireTokenInteractiveParameterBuilder CreateInteractive(IEnumerable<string> scopes)
+        {
+            return AcquireTokenInteractiveParameterBuilder.Create(scopes);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <param name="userAssertion"></param>
+        /// <returns></returns>
+        public static AcquireTokenOnBehalfOfParameterBuilder CreateOnBehalfOf(
+            IEnumerable<string> scopes,
+            UserAssertion userAssertion)
+        {
+            return AcquireTokenOnBehalfOfParameterBuilder.Create(scopes, userAssertion);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <param name="account"></param>
+        /// <returns></returns>
+        public static AcquireTokenSilentParameterBuilder CreateSilent(IEnumerable<string> scopes, IAccount account)
```

Comment:
I'm not sure for this one. 
Also AcquireTokenSilent has 2 overrides (one with a force Boolean parameter)

Reply:
What part aren't you sure of here?  Also, there's a WithForceRefresh() in the AcquireTokenSilentParameterBuilder to cover this overload scenario.

---
In reply to: [246268023](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/766#discussion_r246268023) [](ancestors = 246268023)

Reply:
I think from our discussion we're resolved.  Please reactivate if there are further issues.

---
In reply to: [246500144](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/766#discussion_r246500144) [](ancestors = 246500144,246268023)